### PR TITLE
[#2856] fix(core): Fix the namespace missing issue when get entities from kv storage

### PR DESCRIPTION
**Pull request by jerryshao**
### What changes were proposed in this pull request?

This PR fixes the namespace missing issue when entities are gotten from kv storage.

### Why are the changes needed?

Fix: #2856 

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Modified UTs.

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -539,7 +539,8 @@ tasks.rat {
     "**/LICENSE.*",
     "**/NOTICE.*",
     "ROADMAP.md",
-    "clients/client-python/.pytest_cache/*"
+    "clients/client-python/.pytest_cache/*",
+    "clients/client-python/gravitino.egg-info/*"
```
Comment (yuqi1129) [inline build.gradle.kts:543]: This folder does not exist on my local machine, can anyone help to confirm this point?

**Comment by yuqi1129** [comment]
@jerryshao 

I noticed that you only set the namespace value when deserialization and will not store the value in the key-value store. My question is: Why do we have to change the interface `EntitySerDe` instead of setting it(namespace value) in the caller method?

File: build.gradle.kts (hunk 1 above)
Comment (jerryshao) [reply build.gradle.kts:543]: Try gradle build.

**Comment by jerryshao** [comment]
> @jerryshao
> 
> I noticed that you only set the namespace value when deserialization and will not store the value in the key-value store. My question is: Why do we have to change the interface `EntitySerDe` instead of setting it(namespace value) in the caller method?

You can check the issue to know the details. Simply, it is because the deserialized entity from KV has an empty namespace, but for catalog like kafka needs to get namespace from entity to use, and it cannot get a namespace from context (caller), you can see KafkaOperation's CatalogInfo (CatalogEntity).

Besides, for relational store, they all have namespace set when get from interface, so it is also a behavior alignment.

**Comment by yuqi1129** [comment]
> > @jerryshao
> 
> > 
> 
> > I noticed that you only set the namespace value when deserialization and will not store the value in the key-value store. My question is: Why do we have to change the interface `EntitySerDe` instead of setting it(namespace value) in the caller method?
> 
> 
> 
> You can check the issue to know the details. Simply, it is because the deserialized entity from KV has an empty namespace, but for catalog like kafka needs to get namespace from entity to use, and it cannot get a namespace from context (caller), you can see KafkaOperation's CatalogInfo (CatalogEntity).
> 
> 
> 
> Besides, for relational store, they all have namespace set when get from interface, so it is also a behavior alignment.
> 
> 

I see, let me check again

**Review by yuqi1129 — APPROVED** [review]
It's okay with me.

**Review by mchades — APPROVED** [review]
LGTM

**Pull request merged**